### PR TITLE
Update pki.spec and build.sh

### PR DESCRIPTION
**Pull request by edewata**
The `pki.spec` has been modified to call `build.sh`, and the `build.sh` has been modified to call CMake. This way `build.sh` should work on platforms that do not support RPM, and later on it would be easier to replace CMake with Maven.

### CODE REVIEW

**Comment by edewata** [comment]
Sorry, apparently it's not ready yet.

**Comment by edewata** [comment]
It's ready now.

File: build.sh [hunk 1]
```diff
@@ -390,6 +490,179 @@ if [ "$DEBUG" = true ] ; then
...
+    echo "    $WORK_DIR/dist/pki-tools.jar"
+    echo "    $WORK_DIR/dist/pki-tomcat.jar"
+    echo "    $WORK_DIR/dist/pki-cms.jar"
+    echo "    $WORK_DIR/dist/pki-cmsbundle.jar"
+    echo "    $WORK_DIR/dist/pki-acme.jar"
+    echo "    $WORK_DIR/dist/pki-ca.jar"
+    echo "    $WORK_DIR/dist/pki-kra.jar"
+    echo "    $WORK_DIR/dist/pki-ocsp.jar"
+    echo "    $WORK_DIR/dist/pki-tks.jar"
+    echo "    $WORK_DIR/dist/pki-tps.jar"
```
Comment (fmarco76) [inline build.sh:625]: Are these always built or they depend on the package list?

**Review by fmarco76 — APPROVED** [review]
The overall code looks good. If I understood correctly, the build procedure is still the same as in the documentation. User run the build.sh with rpm and this will call the pki.spec which call the build.sh with different options. Is this correct or I miss something?

**Review by fmarco76 — APPROVED** [review]
LGTM

**Review by ckelleyRH — APPROVED** [review]
LGTM

File: build.sh (hunk 1 above)
Comment (edewata) [reply build.sh]: Good point. They depend on the package list. I've updated the code.

**Comment by edewata** [comment]
@fmarco76 @ckelleyRH Thanks!

So yes, by default the `build.sh` will execute the `dist` target which will create the build artifacts. To install the build supposedly you can run `build.sh install`, but I haven't verified it on a clean machine. To build the RPM you can run `build.sh rpm` which will build the RPM using the spec file, and in the process it will call `build.sh dist` and `build.sh install` with the proper parameters for RPM. I'll update the docs in a separate PR.

**Pull request merged**